lib/core/test/unit/Cardano/Wallet/ApiSpec.hs

```diff
@@ -417,63 +431,64 @@ instance
+    type MkBodyRequest (Verb m s '[ct] a) = [Request]
     gEveryBodyParam _ req =
-        req { requestMethod = reflectMethod (Proxy @m) }
+        [req { requestMethod = reflectMethod (Proxy @m) }]
 
-    type MkHeaderRequest (Verb m s '[ct] a) = Header "Accept" ct -> Request
+    type MkHeaderRequest (Verb m s '[ct] a) = Header "Accept" ct -> [Request]
     gEveryHeader _ req (Header h) =
-        req { requestMethod = reflectMethod $ Proxy @m
-            , requestHeaders = requestHeaders req ++ [ (hAccept, h) ]
-            }
+        [req { requestMethod = reflectMethod $ Proxy @m
+             , requestHeaders = requestHeaders req ++ [ (hAccept, h) ]
+             }]
 instance
     ( ReflectMethod m
     ) => GEveryEndpoints (NoContentVerb (m :: StdMethod))
   where
     gEveryEndpoint _ =
         [defaultRequest { requestMethod = reflectMethod (Proxy @m) }]
 
-    type MkPathRequest (NoContentVerb m) = Request
+    type MkPathRequest (NoContentVerb m) = [Request]
     gEveryPathParam _ req =
-        req { requestMethod = reflectMethod (Proxy @m) }
+        [req { requestMethod = reflectMethod (Proxy @m) }]
 
-    type MkBodyRequest (NoContentVerb m) = Request
+    type MkBodyRequest (NoContentVerb m) = [Request]
     gEveryBodyParam _ req =
-        req { requestMethod = reflectMethod (Proxy @m) }
+        [req { requestMethod = reflectMethod (Proxy @m) }]
 
-    type MkHeaderRequest (NoContentVerb m) = Request
+    type MkHeaderRequest (NoContentVerb m) = [Request]
     gEveryHeader _ req =
-        req { requestMethod = reflectMethod (Proxy @m) }
+        [req { requestMethod = reflectMethod (Proxy @m) }]
+
 
 instance
     ( Wellformed (PathParam t)
     , GEveryEndpoints sub
     ) => GEveryEndpoints (Capture p t :> sub)
   where
     gEveryEndpoint _ =
-        addPathFragment t <$> gEveryEndpoint (Proxy @sub)
+        concatMap (\t -> addPathFragment t <$> gEveryEndpoint (Proxy @sub)) ts
       where
-        t = wellformed :: PathParam t
+        ts = wellformed :: [PathParam t]
 
     type MkPathRequest (Capture p t :> sub) = PathParam t -> MkPathRequest sub
     gEveryPathParam _ req t =
         gEveryPathParam (Proxy @sub) (addPathFragment t req)
 
-    type MkBodyRequest (Capture p t :> sub) = MkBodyRequest sub
+    type MkBodyRequest (Capture p t :> sub) = [MkBodyRequest sub]
     gEveryBodyParam _ req =
-        gEveryBodyParam (Proxy @sub) (addPathFragment t req)
+        fmap (\t' -> gEveryBodyParam (Proxy @sub) (addPathFragment t' req)) t
       where
-        t = wellformed :: PathParam t
+        t = wellformed :: [PathParam t]
 
-    type MkHeaderRequest (Capture p t :> sub) = MkHeaderRequest sub
+    type MkHeaderRequest (Capture p t :> sub) = [MkHeaderRequest sub]
     gEveryHeader _ req =
-        gEveryHeader (Proxy @sub) (addPathFragment t req)
+        fmap (\t' -> gEveryHeader (Proxy @sub) (addPathFragment t' req)) t
       where
-        t = wellformed :: PathParam t
+        t = wellformed :: [PathParam t]
```

Comment:
Would be better to stick to a consistent naming like `ts`  before. These things are actually the same thing so they ought to have the same name.